app/controllers/rails_admin/main_controller.rb

```diff
@@ -56,7 +56,14 @@ def respond_to_missing?(sym, include_private)
     end
 
     def back_or_index
-      params[:return_to].presence && params[:return_to].include?(request.host) && (params[:return_to] != request.fullpath) ? params[:return_to] : index_path
+      params[:return_to].presence && allowed_return_to?(params[:return_to]) && (params[:return_to] != request.fullpath) ? params[:return_to] : index_path
+    end
+
+    def allowed_return_to?(url)
+      return true if url.starts_with?(request.base_url)
+      return false unless url.to_s.start_with?('/')
+
+      !url.to_s.start_with?('//')
```

Comment:
Took inspiration from Rails7's [url_from](https://github.com/rails/rails/blob/cdd14ce1f5196e4bd98df42f89a1cd36ba9d4bee/actionpack/lib/action_controller/metal/redirecting.rb#L203-L212)

Reply:
Tweaking this a bit to account for consolidation suggestion mentioned above, I believe this would cover the cases from the original and your changes.

```suggestion
    def _allowed_return_to?(url)
      url != request.fullpath &&
        (url.starts_with?(request.base_url, '/') && !url.start_with?('//'))
```